### ISSUE TITLE
Not getting partial reads of large parquet file in AWS S3 opened from pre-signed URL

### ISSUE THREAD

**Issue by dude0001** [duckdb/duckdb-wasm#1367]
### What happens?

I am trying to read a large Parquet file from AWS S3 using pre-signed URLs. The expectation is that DuckDB-WASM will be able to do partial reads of the large file. The observed behavior is the entire file is read. I can reproduce this by first registering the file with `registerFileURL` writing a SELECT against the file alias as well as putting the pre-signed URL inline in a SELECT.

### To Reproduce

1. When using `await this.db.registerFileURL('data.parquet', 'https://the-bucket.s3.amazonaws.com/data.parquet?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASI....', DuckDBDataProtocol.HTTP, false);` 

Request 1: HEAD w/ `Range: bytes=0-` header results in 403 w/ `Access-Control-Expose-Headers: Access-Control-Allow-Origin, Accept-Ranges, Content-Range, ETag` header and in the console logs `falling back to full HTTP read for: https://the-bucket.s3.amazonaws.com/data.parquet?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASI...`
This is expected since the pre-signed URL only allows GET

Request 2: GET w/ no relevant header results in 200 reading the entire file

2. When using `select column_1 from 'https://the-bucket.s3.amazonaws.com/data.parquet?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASI...' limit 1;`

Request 1: HEAD w/ no relevant header results in 403 w/ `Access-Control-Expose-Headers: Access-Control-Allow-Origin, Accept-Ranges, Content-Range, ETag` header
Again, this is expected since the pre-signed URL only allows GET

Request 2: GET w/ `Range: bytes=0-0` header results in 206 w/ `Access-Control-Expose-Headers: Access-Control-Allow-Origin, Accept-Ranges, Content-Range, ETag` header and a single byte `P` returned in the Response body.
Seems to prove `Range` header can be used to fetch a partial file.

But then... instead of starting to partially read the data needed to serve the query, what I observed when using  `registerFileURL` seems to repeat.

Request 3: HEAD w/ `Range: bytes=0-` header results in 403 w/ `Access-Control-Expose-Headers: Access-Control-Allow-Origin, Accept-Ranges, Content-Range, ETag` header
This is expected since the pre-signed URL only allows GET.  But why is it doing a HEAD here, when Request 2 showed the server can do partial reads using GET w/ `Range` header?

Request 4: GET w/ no relevant header results in 200 reading the entire file instead of partial reads.

### Browser/Environment:

Chome115.0.5790.171 (Official Build) (64-bit)

### Device:

Lenovo ThinkPad P16

### DuckDB-Wasm Version:

1.27.0

### DuckDB-Wasm Deployment:

shell.duckdb.org

### Full Name:

mark.lambert@gmail.com

### Affiliation:

Episode-VII Solutions

**Comment by dude0001** [comment]
Cross-posted on [Discord](https://discord.com/channels/909674491309850675/921100929984503858/1141211547012702278) in case future readers are looking for more information.

**Comment by dude0001** [comment]
I did see #1366 and this may be related, or have some synergy to refactoring this logic considering these two issues together. I tried testing issue #1367 with v1.24.0 and still have seen the same outcome. The description on #1366 seems to say the issue there does not exist in v1.24.0 and broke in v1.25.0.

I'll see if I can put together a public demo or provide other effort on this.

**Comment by carstonhernke** [comment]
@carlopi Is the relevant chunk of code which performs the detection located here? 
https://github.com/duckdb/duckdb-wasm/blob/91335e0e11703a4f7857c9384e3daca74a11b430/packages/duckdb-wasm/src/bindings/runtime_browser.ts#L177

**Comment by carstonhernke** [comment]
I opened PR #1702 which should resolve this issue :)

**Comment by carlopi** [comment]
Thanks @carstonhernke for the push on this, should be solved.

**Comment by deepakmalik264** [comment]
### Regression: range-detection broken again in v1.30.0+ versions for S3 pre-signed URLs

**Summary:** PR #1717 (shipped in v1.29.0) correctly fixed range-request detection for S3 pre-signed URLs by reading `Content-Range` from the 1-byte GET probe. A refactor between v1.29.0 and v1.30.0 changed that code to read `Content-Length` instead, which silently re-introduced the original bug. v1.30.0, v1.31.0, v1.32.0, and `main` are all affected — duckdb-wasm downloads the entire file on every query against an S3 pre-signed URL.

**Diagnosis:**

In v1.29.0 ([`runtime_browser.ts`](https://github.com/duckdb/duckdb-wasm/blob/v1.29.0/packages/duckdb-wasm/src/bindings/runtime_browser.ts)), the working code was:

```ts
const contentRange = xhr.getResponseHeader('Content-Range')?.split('/')[1];   // ✅ correct
const contentLength2 = xhr.getResponseHeader('Content-Length');
let presumedLength = null;
if (contentRange !== undefined) presumedLength = contentRange;
```

In v1.30.0 ([`runtime_browser.ts`](https://github.com/duckdb/duckdb-wasm/blob/v1.30.0/packages/duckdb-wasm/src/bindings/runtime_browser.ts)) and later, this became:

```ts
let actualContentLength = null;
try { actualContentLength = xhr.getResponseHeader('Content-Length'); } catch (e) {/* ... */}
const contentRange = actualContentLength?.split('/')[1];          // ❌ splits Content-Length, not Content-Range
const contentLength2 = actualContentLength;
let presumedLength = null;
if (contentRange !== undefined) presumedLength = contentRange;
```

Against any standards-compliant server, `Content-Length` for a 206 response is `"1"` (the size of the requested range), not `"bytes 0-0/12345"`. So `"1".split("/")[1]` returns `undefined`, `presumedLength` stays `null`, the success branch is skipped, the HEAD-fallback fires (which returns 403 on pre-signed URLs), and the worker falls through to the full-file download.

**Repro (v1.32.0, Chrome, S3 pre-signed URL with proper CORS allowing `range` and exposing `Content-Length`/`Content-Range`):**

```ts
await db.open({ filesystem: { allowFullHTTPReads: true, reliableHeadRequests: false } });
await db.registerFileURL('data.parquet', presignedS3Url, DuckDBDataProtocol.HTTP, false);
await conn.query("SELECT count(*) FROM 'data.parquet'");
```

Network tab shows:

- ❌ HEAD `Range: bytes=0-` → 403 (expected for pre-signed URL)
- ✅ GET `Range: bytes=0-0` → 206 with `Content-Range: bytes 0-0/N`, `Content-Length: 1`
- ❌ GET (no range) → 200, downloads entire file

**Fix:** Restore reading `Content-Range` directly (one-line revert of the offending change). Happy to put up a PR if helpful — let me know.

**Workaround:** Pin to `1.29.0` exactly. Tested and working against S3 pre-signed URLs.

cc @carlopi @carstonhernke @dude0001 @Fil @llimllib
